opentelemetry-sdk/src/logs/logger_provider.rs

```diff
@@ -346,9 +338,43 @@ mod tests {
             Ok(())
         }
     }
+
+    #[derive(Debug, Clone)]
+    struct TestProcessorForResource {
+        resource: Arc<Mutex<Resource>>,
+    }
+    impl LogProcessor for TestProcessorForResource {
+        fn emit(&self, _data: &mut SdkLogRecord, _scope: &InstrumentationScope) {
+            // nothing to do.
+        }
+
+        fn force_flush(&self) -> OTelSdkResult {
+            Ok(())
+        }
+
+        fn shutdown(&self) -> OTelSdkResult {
+            Ok(())
+        }
+
+        fn set_resource(&mut self, resource: &Resource) {
+            let mut res = self.resource.lock().unwrap();
+            *res = resource.clone();
+        }
+    }
+    impl TestProcessorForResource {
+        fn new() -> Self {
+            TestProcessorForResource {
+                resource: Arc::new(Mutex::new(Resource::empty())),
+            }
+        }
+        fn resource(&self) -> Resource {
+            self.resource.lock().unwrap().clone()
+        }
+    }
+
     #[test]
     fn test_logger_provider_default_resource() {
```

Comment:
This test should either include the assertions to check resources for exporter as well or let's rename the test to: 
```suggestion
    fn test_logger_provider_default_resource_for_log_processor() {
```

Reply:
Basically, we need to test that both processors and exporter are able to get hold of the provider's `Resource`.

Reply:
Provider's responsibility ends when it hands over Resource to processor. Then it is upto processor to pass it to exporter. So this test is testing provider only.

I'll add tests to cover exporter too. Do you prefer to modify the existing test itself? or add one in process.rs file?

Reply:
added exporter coverage too to this test